client/src/pages/LibrarySearchPage/LibrarySearchPage.js

```diff
@@ -106,7 +130,11 @@ class LibrarySearchPage extends Component {
         activeFilter: event.target.getAttribute('value'),
       },
       function() {
-        this.searchLibrary({ q: this.state.searchString });
+        if (this.state.activeFilter === 'all') {
```

Comment:
You will want to add one more check here: you should add `|| this.state.searchString.length > 0` to this `if` condition. The `searchLibrary` function should be called whenever a search string has been provided, and `fetchTableDataByType` should only be called when the search string is empty. `fetchTableDataByType` returns all documents, whereas `searchLibrary` returns the documents filtered by data type (if selected) and the search string.